cynic-parser/src/printing/type_system.rs

```diff
@@ -87,6 +87,17 @@ impl<'a> Pretty<'a, Allocator<'a>> for NodeDisplay<SchemaDefinition<'a>> {
 
         builder = builder.append(allocator.text("schema"));
 
+
+        let mut directives = self.0.directives().peekable();
+        let mut directives_pretty = allocator.nil();
+        if directives.peek().is_some() {
+            directives_pretty = allocator
+                .space()
+                .append(allocator.intersperse(directives.map(NodeDisplay), allocator.softline()));
+        }
+
+        builder = builder.append(directives_pretty);
```

Comment:
Is this code generated? I wrote this bit by hand, I haven't looked very deep, will do some more exploration.

Reply:
No, this is all hand written.  Getting the output right has been a challenge, not sure if it'd be worth the effort to automate.